gpflow/base.py

```diff
@@ -107,7 +107,7 @@ def __init__(
 
         self._transform = transform
         self.prior = prior
-        self.prior_on = prior_on
+        self._prior_on = PriorOn(value)
```

Comment:
mypy currently complains about the former code. it's an open ticket in mypy to allow it. for now, i've just changed it

Reply:
the alternative is a `# type: ignore`